### PR TITLE
chore(release): pull release/v1.55.0 into main

### PR DESCRIPTION
**Pull request by devops-github-rudderstack**
:crown: *An automated PR*

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3057/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3057/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3057/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3057/test-report.html

**Review by aashishmalik — APPROVED** [review]

File: src/cdk/v2/destinations/intercom/utils.js [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+  }
+  requestBody.companies = [companyObj];
+  const response = defaultRequestConfig();
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = `${BASE_ENDPOINT}/users`;
+  response.headers = {
+    'Content-Type': JSON_MIME_TYPE,
+    Authorization: `Bearer ${Config.apiKey}`,
+    Accept: JSON_MIME_TYPE,
+    'Intercom-Version': '1.4',
```
Comment (sanpj2292) [inline src/cdk/v2/destinations/intercom/utils.js:182]: [Q] Is the hardcoding of version expected ? @mihir-4116

Comment (sanpj2292) [reply src/cdk/v2/destinations/intercom/utils.js:182]: In `getHeaders` we are using version check and putting a version value in `Intercom-version` header. Want to confirm if this is expected.

File: test/deleteUsers/data/iterable/nw_client_data.json
Comment (krishna2020) [inline test/deleteUsers/data/iterable/nw_client_data.json:1]: Have these got migrated to any other format ?

File: src/cdk/v2/destinations/intercom/utils.js (hunk 1 above)
Comment (mihir-4116) [reply src/cdk/v2/destinations/intercom/utils.js:182]: yes, it is expected as it's old version code so Intercom-Version should be 1.4

File: test/deleteUsers/data/iterable/nw_client_data.json
Comment (sanpj2292) [reply test/deleteUsers/data/iterable/nw_client_data.json:1]: Seems like this got deleted in https://github.com/rudderlabs/rudder-transformer/pull/3057/commits/de4e321a58a3e93d5592b8e9a967334685af26d4

**Review by krishna2020 — APPROVED** [review]

File: test/deleteUsers/data/iterable/nw_client_data.json
Comment (krishna2020) [reply test/deleteUsers/data/iterable/nw_client_data.json:1]: File history doesn't seem to be clean - It got deleted even before it - https://github.com/rudderlabs/rudder-transformer/pull/2672

Comment (krishna2020) [reply test/deleteUsers/data/iterable/nw_client_data.json:1]: I see only this one file got added in PR #2672 & rest all similar files deleted

**Review by mihir-4116 — APPROVED** [review]

File: test/deleteUsers/data/iterable/nw_client_data.json
Comment (chandumlg) [reply test/deleteUsers/data/iterable/nw_client_data.json:1]: I deleted the old unused deleteUsers test cases data. They are not being used.

**Pull request merged**